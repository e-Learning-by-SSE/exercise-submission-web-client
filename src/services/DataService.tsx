import React from "react";
import SparkyAuthentifikation from "./SparkyAuthentifikation";
import Stumgmtbackend from "./Stumgmtbackend";
import { AssignmentDto, UserDto } from "stumgmtbackend";
import { SubmissionClient } from "./SubmissionClient";
import { ErrorInterface } from "../interface/ErrorInterface";


export default class DataService {

    private accesstoken:string | null;
    private user: UserDto | null = null;

    constructor() {
        this.accesstoken = localStorage.getItem("token");
        if(this.checkToken()) {
            this.getCurrentUser().then((user) => {
                this.user = user;
            });
        }

    }

    private checkToken(): boolean {
        if (this.accesstoken != null) {
            
        }
        return true;
    }

    public getStumgmtbackend(): Stumgmtbackend {
        if (this.accesstoken == null) {
            throw new Error("No access token");
        }
        return new Stumgmtbackend(this.accesstoken);
    }

    public getSubmissisonClient(): SubmissionClient {
        if (this.accesstoken == null) {
            throw new Error("No access token");
        }
        return new SubmissionClient(this.accesstoken);
    }

    private async getCurrentUser(): Promise<UserDto> {
        let client = this.getStumgmtbackend();
        let response = await client.getCurrentUser();
        return response;
    }

    public async getGroupName(assignment: AssignmentDto): Promise<string> {
        let user = await this.getCurrentUserDto();
        if(assignment.collaboration) {
            let stumgmt = this.getStumgmtbackend();
            let group = await stumgmt.getGroup(assignment.id, user.id);
            return group.name;
        } else {
            return user.username;
        }
    }

    public async getCurrentUserDto(): Promise<UserDto> {
        let user = this.user;
        if(user == null) {
            user = await this.getCurrentUser();
        }
        return user;
    }

}