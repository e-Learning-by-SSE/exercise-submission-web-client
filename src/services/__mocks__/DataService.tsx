import { UserDto, UserDtoRoleEnum } from "stumgmtbackend";

export default class DataService {
    constructor() {
    }

    public async getCurrentUserDto(): Promise<UserDto> {
        return new Promise<UserDto>((resolve, reject) => {
            resolve({
                username: "test",
                 id: "1",
                 role: UserDtoRoleEnum.USER,
                displayName: "test"
            });
        });
    }
}