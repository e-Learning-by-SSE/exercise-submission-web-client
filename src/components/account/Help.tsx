import { queryByDisplayValue } from "@testing-library/react";
import React from "react";
import { Header, Modal, Grid, Placeholder, Button, Icon } from "semantic-ui-react";
import DataService from "../../services/DataService";

export default class Help extends React.Component<React.PropsWithChildren<{ onClosed: () => void}>, {open: boolean, informationContent: any}> {
    constructor(props: React.PropsWithChildren<{onClosed: () => void}>) {
        super(props);
        this.state = {open: true, informationContent: this.createInformationPlaceholder()};
    }

    private createInformationPlaceholder():  React.ReactElement<any, string | React.JSXElementConstructor<any>> {
        
        return (<><Grid.Row>
                <Placeholder>
                    <Placeholder.Header>
                        <Placeholder.Line />
                    </Placeholder.Header>
                </Placeholder>
            </Grid.Row><Grid.Row>
                <Placeholder>
                    <Placeholder.Header>
                        <Placeholder.Line />
                    </Placeholder.Header>
                </Placeholder>
            </Grid.Row><Grid.Row>
                <Placeholder>
                    <Placeholder.Header>
                        <Placeholder.Line />
                    </Placeholder.Header>
                </Placeholder>
            </Grid.Row><Grid.Row>
                <Placeholder>
                    <Placeholder.Header>
                        <Placeholder.Line />
                    </Placeholder.Header>
                </Placeholder>
            </Grid.Row></>);
    }


    private createInformationContent() {
        let api = new DataService();
        let user = api.getCurrentUserDto().then((user) => {
                this.setState({informationContent: <><Grid.Row>
                    <Header size="small">{user.username}</Header>
                </Grid.Row><Grid.Row>
                        <Header size="small">{user.role}</Header>
                    </Grid.Row><Grid.Row>
                        <Header size="small">{user.role}</Header>
                    </Grid.Row><Grid.Row>
                        <Header size="small">True</Header>
                    </Grid.Row></>}); 
        });

    }

    private createHelpModal() : React.ReactElement<any, string | React.JSXElementConstructor<any>> {


        return (
            <Modal open= {this.state.open}>
                <Modal.Header>Help-Menu</Modal.Header>
                <Modal.Content>
                    <Modal.Description>
                        <Header>User-Informations</Header>
                        <Grid columns={2} divided>
                            <Grid.Column>
                                <Grid.Row>
                                    <Header size="medium">Username:</Header>
                                </Grid.Row>
                                <Grid.Row>
                                    <Header size="medium">Role:</Header>
                                </Grid.Row>
                                <Grid.Row>
                                    <Header size="medium">Group:</Header>
                                </Grid.Row>
                                <Grid.Row>
                                    <Header size="medium">Connected:</Header>
                                </Grid.Row>
                            </Grid.Column>
                            <Grid.Column>
                             {this.state.informationContent}
                            </Grid.Column>
                        </Grid>
                    </Modal.Description>
                </Modal.Content>
                <Modal.Actions>
                    <Button icon color='red' onClick={() =>{ this.setState({open: false}); this.props.onClosed()}}>
                        <Icon name='close' />
                        Close
                    </Button>
                </Modal.Actions>

            </Modal>

            
        )
    }
  
    
    render() {
        this.createInformationContent();
            return (
                <div className="help">
                     {this.createHelpModal()}
                </div>
            );
   
        }
    }