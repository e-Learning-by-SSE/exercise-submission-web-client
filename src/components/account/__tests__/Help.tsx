import renderer from "react-test-renderer";
import Help from "../Help";
import DataService from "../../../services/DataService";
import { act, render, screen, waitFor } from "@testing-library/react";


jest.mock("../../../services/DataService");


it("render Help", async ()  => {
    
   
    const account = render(<Help onClosed={() => {}}/>);

    await waitFor(() => screen.getByText("test"));

    expect(account).toMatchSnapshot();
    
    
});
