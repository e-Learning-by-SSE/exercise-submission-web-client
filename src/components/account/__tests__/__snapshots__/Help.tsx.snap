// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Help 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="dimmable dimmed"
  >
    <div>
      <div
        class="help"
      />
    </div>
    <div
      class="ui page modals dimmer transition visible active"
      style="display: flex;"
    >
      <div
        class="ui modal transition visible active"
      >
        <div
          class="header"
        >
          Help-Menu
        </div>
        <div
          class="content"
        >
          <div
            class="description"
          >
            <div
              class="ui large header"
            >
              User-Informations
            </div>
            <div
              class="ui divider"
            />
            <div
              class="ui divided two column grid"
            >
              <div
                class="row"
              >
                <div
                  class="column"
                >
                  <div
                    class="ui medium header"
                  >
                    Username:
                  </div>
                </div>
                <div
                  class="column"
                >
                  <div
                    class="ui small header"
                  >
                    test
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="column"
                >
                  <div
                    class="ui medium header"
                  >
                    Role:
                  </div>
                </div>
                <div
                  class="column"
                >
                  <div
                    class="ui small header"
                  >
                    USER
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="column"
                >
                  <div
                    class="ui medium header"
                  >
                    Group:
                  </div>
                </div>
                <div
                  class="column"
                >
                  <div
                    class="ui small header"
                  >
                    USER
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="column"
                >
                  <div
                    class="ui medium header"
                  >
                    Connected:
                  </div>
                </div>
                <div
                  class="column"
                >
                  <div
                    class="ui small header"
                  >
                    True
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="actions"
        >
          <button
            class="ui red icon button"
          >
            <i
              aria-hidden="true"
              class="close icon"
            />
            Close
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="help"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
